From d8056765d736b7143531d6205ed2327474ea9a2a Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Tue, 16 Oct 2018 16:47:24 +0800
Subject: [PATCH 3/4] driver: usb: xhci: fix compiler warning about parameter
 type mismatch

vers/usb/host/xhci.c: In function 'xhci_gen_setup':
drivers/usb/host/xhci.c:4918:18: warning: format '%x' expects argument of type 'unsigned int', but argument 5 has type 'u64 {aka long long unsigned int}' [-Wformat=]
  xhci_info(xhci, "hcc params 0x%08x hci version 0x%x quirks 0x%08x\n",
                  ^
     xhci->hcc_params, xhci->hci_version, xhci->quirks);
                                          ~~~~~~~

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/usb/host/xhci.c |    2 +-
 drivers/usb/host/xhci.h |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/host/xhci.c b/drivers/usb/host/xhci.c
index 0df54df..442d2d9 100644
--- a/drivers/usb/host/xhci.c
+++ b/drivers/usb/host/xhci.c
@@ -4915,7 +4915,7 @@ int xhci_gen_setup(struct usb_hcd *hcd, xhci_get_quirks_t get_quirks)
 		return retval;
 	xhci_dbg(xhci, "Called HCD init\n");
 
-	xhci_info(xhci, "hcc params 0x%08x hci version 0x%x quirks 0x%08x\n",
+	xhci_info(xhci, "hcc params 0x%08x hci version 0x%x quirks 0x%llx\n",
 		  xhci->hcc_params, xhci->hci_version, xhci->quirks);
 
 	return 0;
diff --git a/drivers/usb/host/xhci.h b/drivers/usb/host/xhci.h
index e9dfc3d..2f21382 100644
--- a/drivers/usb/host/xhci.h
+++ b/drivers/usb/host/xhci.h
@@ -1822,7 +1822,7 @@ struct xhci_hcd {
 #define XHCI_SSIC_PORT_UNUSED	(1 << 22)
 #define XHCI_NO_64BIT_SUPPORT	(1 << 23)
 #define XHCI_MISSING_CAS	(1 << 24)
-#define XHCI_DIS_U1U2_WHEN_U3 (1 << 32)
+#define XHCI_DIS_U1U2_WHEN_U3 (1UL << 32)
 #define XHCI_STOP_EP_IN_U1     BIT(34)
 /* For controller with a broken Port Disable implementation */
 #define XHCI_BROKEN_PORT_PED	(1 << 25)
-- 
1.7.9.5

